src/fromager/candidate.py

```diff
@@ -43,6 +43,20 @@ def __repr__(self) -> str:
             return f"<{self.name}=={self.version}>"
         return f"<{self.name}[{','.join(self.extras)}]=={self.version}>"
 
+    def __hash__(self) -> int:
+        return hash(self.__repr__())
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Candidate):
+            return (
+                self.name == other.name
+                and self.version == other.version
+                and self.build_tag == other.build_tag
+                and self.extras == other.extras
+                and self.is_sdist == other.is_sdist
+            )
+        return False
```

Comment:
You have to return NotImplemented of you don't know how to compare an object.
```suggestion
        return NotImplemented
```

Reply:
Won't need this anymore